### DISCUSSION TITLE
gattc_discover_descriptors stuck, no event raised

### DISCUSSION THREAD

**Discussion by amplatzer** [micropython/micropython#16099] [General]
Hi Team,

peripheral chip:  Amiccom A7107 
master: ESP32-S3R2@ Mpy: 1.23.0

There is a BLE peripheral that I'm able to connect and subscribe with the NRFconnect or lightblue APP. 
but with ESP32 client (@ micropython 1.23.0), after I discover all characteristics, I call ble.gattc_discover_descriptors(3,80,85) .

I was expecting  _IRQ_GATTC_DESCRIPTOR_RESULT event, so I can subscribe to the peripheral by changing value of the descriptor 0x2902, but nothing. 

the code stuck, until the peripheral disconnect from ESP32, and event _IRQ_GATTC_DESCRIPTOR_DONE was finally raised. 

Any idea why no DESCRIPTOR was found? 
PS: The code works fine with other similar peripherals except this particular one.

**Comment by bidrohini** [comment]
Try to verify with a BLE scanner (like nRF Connect) whether the 0x2902 descriptor is actually present for that characteristic.

**Reply by amplatzer** [reply]
> Try to verify with a BLE scanner (like nRF Connect) whether the 0x2902 descriptor is actually present for that characteristic.

thanks for the response, like I mentioned,  I'm able to connect and subscribe with the NRFconnect or Lightblue APP..

**Reply by amplatzer** [reply]
![IMG_9586](https://github.com/user-attachments/assets/6110ba30-d97a-42db-b093-34d1c7653206)
